### ISSUE TITLE
Extending scripting JS function API

### ISSUE THREAD

**Issue by edzius** [gerbera/gerbera#877]
Hi guys,

I'm very happy to see this project evolving and you guys doing great job get it moving.
However i've missed flexibility and some functionality classifying added media entries, e.g. by year, by rating, etc. This information could be acquired from external sources so i'd like to propose and get your opinion on extending duktape's exposed JS API with:

* possibility to do HTTP GET -- initially could be plain HTTP support; synchronous call.   
* possibility to do HTTP POST -- just for HTTP API consistency; basically HTTP GET should do all heavy lifting. 
* possibility to access host ENV -- for wider scripts configuration capabilities.   

I don't know of you plans for future scripting support, however this proposition is based on assumption you stick with duktape integration. 

Please let me know your opinion on this. Thanks!

**Comment by ytimenkov** [comment]
I had same goals in #747.

Extending interaction via HTTP is also possible, however on its own it has some corner cases, for example if service was unavailable at the time of import? What about timeouts?

Also interchange via files makes it more straightforward to backup / repopulate this info as well as implement HTTP fetch logic in any language you want.

Say you want to fetch fanart? Then you need to extend Gerbera JS with arbitrary file access API as well...

**Comment by edzius** [comment]
Yes, introducing networking to retrieve meta data has it's own risks. Right now i can foresee following issues:

1. Network resource availability (remote host is dead, routes problem, network stack problems, etc.)

Handling such cases would mean we need couple more mechanisms to defer imports and to expire deferred imports, e.g. if remote host "never" gets online. Deferred imports and whole machinery it brings sounds like a big change and might not be introduced at once. On the other hand such metadata fetch rules might not need to be part of gerbera core; instead it could be import's script's responsibility to handle failing fetch requests. In short gerbera core provides tools, and import scripts holds whole application rules.

2. Synchronous fetch operation (when remote resource is not available)

To deal with it "the right way" we would need asynchronous fetch, which would mean bigger changes for media import internals are required; maybe deferred imports mentioned above could benefit here. However for sake of simplicity maybe for now simply configurable request timeout would be sufficient?

3. Fetch spoofing

Theoretically possible to spoof fetch replies especially if public network resources are used, however as far as i see while there is no access to file system IO, it should not compromise system's security.

Anyway to summarize all the problems seems to be bound to gerbera deployment, thus gerbera user/maintainer/administrator should be aware about the risks of using network resources. Also in this case good practice could be to setup metadata server on local host or local network, and this could solve most problems related to service availability, latency and security. 

> Also interchange via files makes it more straightforward to backup / repopulate this info as well as implement HTTP fetch logic in any language you want.

I'm not sure if i understood what you mean by saygin `interchange via files`? 

> Say you want to fetch fanart? Then you need to extend Gerbera JS with arbitrary file access API as well...

Well i'm more concerned about textual metadata and media sorting/grouping, and less about augmenting media with additional data.
